src/EntityFramework/Core/Query/PlanCompiler/TransformationRulesContext.cs

```diff
@@ -262,6 +262,49 @@ internal bool IsScalarOpTree(Node node, Dictionary<Var, int> varRefMap)
             return IsScalarOpTree(node, varRefMap, ref nodeCount);
         }
 
+        /// <summary>
+        /// Is this tree uses user-defined functions
+        /// Simplifing query with UDFs could caused to suboptimal plans
+        /// </summary>
+        /// <param name="node"> Current subtree to process </param>
+        /// <param name="varMap">Mapped variables</param>
+        /// <returns></returns>
+        internal bool IncludeCustomFunctionOp(Node node, Dictionary<Var, Node> varMap)
+        {
+            PlanCompiler.Assert(varMap != null, "Null varRef map");
+
+            if (node.Op.OpType == OpType.VarRef)
+            {
+                var varRefOp = (VarRefOp)node.Op;
+                if (varMap.TryGetValue(varRefOp.Var, out var newNode))
+                {
+                    if (IncludeCustomFunctionOp(newNode, varMap))
```

Comment:
Collapse the whole `if`:

```suggestion
                    return IncludeCustomFunctionOp(newNode, varMap);
```

Reply:
BTW, the my suggestion involves deleting a few subsequent lines but GitHub doesn't allow suggestions to do that yet.

Reply:
Fixed.